R/mod_edish.R

```diff
@@ -492,6 +492,17 @@ check_mod_edish <- function(
     lb_result_var, ref_range_upper_lim_var, receiver_id,
     warn, err
   )
+  
+  # Check only if `arm_default_vals` is a character vector
+  # Reason: Arbitrary values allowed in case multiple studies are included with different arm values
+  CM$assert(
+    container = err,
+    cond = is.character(arm_default_vals) | is.null(arm_default_vals),
```

Comment:
The condition on the suggestion above is still softer than the one at the beginning of `edish_server` (on line 161 on this file). If the caller uses `character(0)` or an `""` as the value of `arm_default_vals`, this check passes, but the one on `edish_server` makes the application fail a bit mysteriously. My suggestion is to take the assertion from 161 and transform into a boolean test (from `assert_character` into `test_character`) here:
```suggestion
    cond = checkmate::test_character(
      arm_default_vals, min.chars = 1, any.missing = FALSE, all.missing = FALSE, unique = TRUE, min.len = 1, 
      null.ok = TRUE
    ),
```